doc/udev-rules.org

```diff
@@ -0,0 +1,244 @@
+* Udev Rules in yast-network
+
+https://github.com/yast/yast-network
+
+** What are Udev Rules
+
+If a computer has two network cards, how does it determine which one is *eth0*
+and which one is *eth1*?
+
+A Linux machine may have several network interfaces, such as eth0, eth1,
+wlan0, enp0s3.
+These udev rules ensure the correct assignment of interface names.
+
+_(
+  "<p><b>Udev Rules</b> are rules for the kernel device manager that allow\n" \
+    "associating the MAC address or BusID of the network device with its name (for\n" \
+    "example, eth1, wlan0 ) and assures a persistent device name upon reboot.\n"
+)
+
+For a NIC a widget exists where you can change its name and choose whether it
+will be pinned to its MAC (Ethernet address), or its PCI bus address.
+
+** Why
+
+The code of yast-network is messy, so Udev Rules is one area to clean it up.
+It is simple enough and well defined.
+
+** Plan
+
+- [x] collect all mentions of "udev" in yast-network
+- [ ] read them all and design a decent API
+- [ ] cover the code with tests
+- [ ] use the new API, possibly using old code underneath
+- [ ] replace old implementation with a better one
```

Comment:
let me phrase it a bit more officially.
```
- [ ] collect existing requirements
- [ ] propose new API that fullfils this requirements
- [ ] write tests to ensure existing code works as initially expected
- [ ] replace code with new one
```
I do not think we should necessary use old implementation if new one will be straight enough